agent/lib/kontena/service_pods/lifecycle_hook_manager.rb

```diff
@@ -0,0 +1,93 @@
+module Kontena
+  module ServicePods
+    class LifecycleHookManager
+      include Kontena::Helpers::RpcHelper
+      include Common
+
+      attr_reader :service_pod
+
+      # @param [ServicePod] service_pod
+      def initialize(service_pod)
+        @service_pod = service_pod
+      end
+
+      # @return [Boolean]
+      def on_pre_start
+        pre_start_hooks = hooks_for('pre_start')
+        return if pre_start_hooks.size == 0
+
+        service_config = config_container(service_pod.dup)
+        service_config['Entrypoint'] = '/bin/sh'
+        service_config['Cmd'] = ['-c', 'sleep 600']
```

Comment:
But yes, there is a race cond between exec and weave attach.. not sure how we can fix that?